index.js

```diff
@@ -53,11 +52,11 @@ function fastProxy (opts = {}) {
 
       let body = null
       // according to https://tools.ietf.org/html/rfc2616#section-4.3
-      // proxy should ignore message body when it's a GET or HEAD request
+      // proxy should ignore message body when it's a GET, HEAD or OPTIONS request
       // when proxy this request, we should reset the content-length to make it a valid http request
-      if (req.method === 'GET' || req.method === 'HEAD') {
-        if (headers['content-length']) {
-          headers = filterHeaders(headers, 'content-length')
+      if (req.method === 'GET' || req.method === 'HEAD' || req.method === 'OPTIONS') {
+        if (headers.hasOwnProperty('content-length')) {
+          delete headers['content-length']
```

Comment:
Here's the context when it was introduced https://github.com/fastify/fast-proxy/pull/38 (for the same undici reason I raised about OPTIONS)

-We could think it was made that way to prevent object mutation but headers are taken from a spread, so no risks of modifying elsewhere, and `headers` object is already mutated
```
headers['x-forwarded-host'] = headers.host
headers.host = url.hostname
```
-`filterHeaders` ensures a case insensitive removal, but first, headers are supposed to already be lowercase and anyway, a case-sensitive comparison is made before calling it.

Are you ok with delete then ?
But I can obviously leave it as it was, the priority being to fix that OPTIONS issue.

Reply:
delete is usually slower than just creating a new Object.
Maybe just fix the OPTIONS and keep that code and just open a new issue to discuss it.